ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-blocks/embeddedActivityBlock/embeddedActivityBlock.component.ts

```diff
@@ -1,63 +1,91 @@
-  @Input() validationRequested: boolean;
-  @Input() studyGuid: string;
-  @Output() componentBusy = new EventEmitter<boolean>(true);
-
-  public activity: ActivityForm;
-  private readonly LOG_SOURCE = 'EmbeddedActivityBlockComponent';
-
-  constructor(private activityServiceAgent: ActivityServiceAgent,
-              private changeDetector: ChangeDetectorRef,
-              private submitAnnouncementService: SubmitAnnouncementService,
-              private logger: LoggingService) { }
-
-  ngOnInit(): void {
-    this.getActivity();
-    this.submitAnnouncementService.submitAnnounced$.pipe(
-        tap(() => this.componentBusy.emit(true)),
-        concatMap(() => this.activityServiceAgent.flushForm(this.studyGuid, this.instance.instanceGuid)),
-        catchError(err => {
-            this.logger.logError(this.LOG_SOURCE, 'An error during completing an activity', err);
-            return EMPTY;
-        }),
-        take(1),
-        finalize(() => this.componentBusy.emit(false))
-    ).subscribe();
-  }
-
-  public getSectionId(index: number, {name}: ActivitySection): string {
-    return `${index}_${name}`;
-  }
-
-  private getActivity(): void {
-    this.activityServiceAgent.getActivity(of(this.studyGuid), of(this.instance.instanceGuid))
-      .pipe(
-        catchError(err => {
-          this.logger.logError(this.LOG_SOURCE, 'An error during getting a full activity', err);
-          return EMPTY;
-        }),
-        take(1))
-      .subscribe((activity: ActivityForm) => {
-        this.activity = activity;
-        this.changeDetector.detectChanges();
-      });
-  }
+    @Input() instance: ActivityInstance;
+    @Input() readonly: boolean;
+    @Input() validationRequested: boolean;
+    @Input() studyGuid: string;
+    @Output() componentBusy = new EventEmitter<boolean>(true);
+    @Output() deletedActivity = new EventEmitter<string>();
+
+    public activity: ActivityForm;
+    private readonly LOG_SOURCE = 'EmbeddedActivityBlockComponent';
+
+    constructor(private activityServiceAgent: ActivityServiceAgent,
+                private changeDetector: ChangeDetectorRef,
+                private submitAnnouncementService: SubmitAnnouncementService,
+                private logger: LoggingService) {
+    }
+
+    ngOnInit(): void {
+        this.getActivity();
+        this.flushFormOnSubmit();
+    }
+
+    public getSectionId(index: number, {name}: ActivitySection): string {
+        return `${index}_${name}`;
+    }
+
+    public deleteInstance(): void {
+        this.activityServiceAgent.deleteActivityInstance(this.studyGuid, this.instance.instanceGuid).pipe(
+            catchError(err => {
+                this.logger.logError(this.LOG_SOURCE, 'An error during deleting an activityInstance', err);
+                return EMPTY;
+            }),
+            take(1)
+            )
+            .subscribe(() => {
+                this.deletedActivity.emit(this.instance.instanceGuid);
+            });
```

Comment:
Would suggest adding the componentBusy.emit(true)/componentBusy.emit(false) like we have in the "flushFormOnSubmit" method. Prevent other stuff from happening until these operations complete

Reply:
Had not noticed before, but we should have same mechanism of notifying deletion in the modal component. (emitter and wrapping the deletion call with busy true/busy false.

Reply:
Added at 514082d258f6237d04c5f73494a3fe96c81dc38c.